### DISCUSSION TITLE
how to organize formula that relies on platform-independent data files that themselves require building?

### DISCUSSION THREAD

**Discussion by aryairani** [Homebrew/discussions#5105] [Writing Formulae/Casks]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I've got Haskell project `A`, a command-line tool with a built-in web service, and Node project `B`, which generates the static html/js resources for the built-in web service  in `A`.

The output of `B`, since it's just html/css/js files, doesn't change based on which OS or architecture it's built with.

Option 1: Currently we have a formula for `A` that treats `B` as a resource that's retrieved and staged during the built of `A`, but that seemed wasteful to me — `A` _does_ need to be built for each OS and architecture, so this ends up building `B` multiple times as well — once for each bottle of A, which seem somewhat wasteful.

Option 2: I make a separate formula for `B`, and then `A`'s formula `depends_on "B"`; I reference the specific files as `(Formula["B"].opt_share/"b-stuff")` in a launcher script for `A`.  But even though `brew --build-bottle` detects that `B` is relocatable, it doesn't detect that it's platform independent, so I still would end up having to build `B` multiple times for no clear reason other than "because".

**Comment by SMillerDev** [comment]
If it doesn't register as platform independent that means it's detecting something specific for the platform. Making a pull request would be the only real way to debug that

**Reply by aryairani** [reply]
Hi @SMillerDev Ok thanks — well ok actually two questions — as I'm rereading my question and your answer, I'm not sure how I determined that "it doesn't detect that it's platform independent", maybe it just listed the OS in the bottle block that was output?

So question 1: Is there actually such a thing as a platform-independent bottle, and what does that look like in the bottle block? Is it just missing the `arm64_sonoma:` key(is this a key? I don't really know ruby) on the `sha256` line? or is there a key that means platform-independent?

And question 2: Would it be okay homebrew etiquette to create the pull request just to debug that? I'm guessing yes from your answer; but would have assumed not, otherwise. :) I can do that.

**Reply by aryairani** [reply]
And then apart from the question of possibly-duplicative bottles, did you also have any feedback about whether Option 1 vs Option 2 is better? Or some other option? Thanks!!

**Reply by SMillerDev** [reply]
> So question 1: Is there actually such a thing as a platform-independent bottle, and what does that look like in the bottle block? Is it just missing the arm64_sonoma: key(is this a key? I don't really know ruby) on the sha256 line? or is there a key that means platform-independent?

The key would be `:all` in that case.

> And question 2: Would it be okay homebrew etiquette to create the pull request just to debug that? I'm guessing yes from your answer; but would have assumed not, otherwise. :) I can do that.

Sure.

**Reply by aryairani** [reply]
I tried creating the PR here, but CI was cancelled due to the github repo for the static resources having few stars. https://github.com/Homebrew/homebrew-core/pull/161979 😅
